### PR TITLE
Some small modernization changes in ecl classes

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: ebos/eclalugridvanguard.hh [hunk 1]
```diff
@@ -170,17 +170,17 @@ public:
...
-                                                    this->cartesianIndexMapper(),
-                                                    this->grid(),
-                                                    this->cellCentroids(),
-                                                    getPropValue<TypeTag,
-                                                    Properties::EnableEnergy>(),
-                                                    getPropValue<TypeTag,
-                                                    Properties::EnableDiffusion>()));
-        // Re-ordering  for ALUGrid
-        globalTrans_->update(false, [&](unsigned int i) { return gridEquilIdxToGridIdx(i);});
+           globalTrans_.reset(new TransmissibilityType(this->eclState(),
```
Comment (bska) [inline ebos/eclalugridvanguard.hh:173]: If you're changing this anyway, how about switching to `make_unique<TransmissibilityType>()` instead?

Comment (akva2) [reply ebos/eclalugridvanguard.hh]: sure.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks.  This looks good to me and I'll merge into master.

**Pull request merged**